scripts/release_info.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Script that simplifies releases by collecting various information.
+Should be run from the oppia root dir.
+"""
+import collections
+import os
+import re
+import subprocess
+import sys
+
+
+class ChangedBranch(object):
+    def __init__(self, new_branch):
+        get_branch_cmd = 'git symbolic-ref -q --short HEAD'.split()
+        self.old_branch = subprocess.check_output(get_branch_cmd).strip()
+        self.new_branch = new_branch
+        self.is_same_branch = self.old_branch == self.new_branch
+
+    def __enter__(self):
+        if not self.is_same_branch:
+            try:
+                subprocess.check_output(["git", "checkout", self.new_branch])
+            except subprocess.CalledProcessError:
+                print ('\nCould not change to %s. This is most probably because'
+                       ' you are in a dirty state' % self.new_branch)
+                sys.exit(1)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if not self.is_same_branch:
+            subprocess.check_output(["git", "checkout", self.old_branch])
+
+
+Log = collections.namedtuple('Log', ['sha1', 'author', 'email', 'message'])
+
+
+def _get_current_version():
+    """Get the newest tag in the current commits branch (the version number).
+
+    Returns (str): Latest version tag
+    """
+    git_describe_cmd = 'git describe --abbrev=0'.split()
+    return subprocess.check_output(git_describe_cmd).strip()
+
+
+def _gather_logs(start, stop="HEAD"):
+    """Gathers the logs between the start and endpoint.
+
+    Args:
+        start (str): Tag, Branch or SHA1 of start point
+        stop (str):  Tag, Branch or SHA1 of end point, defaults to HEAD
+
+    Returns:
+        list[Log]: List of Logs
+
+    """
+    group_sep = '\x1D'
+    git_log_cmd = \
```

Comment:
We never use "\" at the end of a line, because it's hard to tell what it's escaping (it may be a space after it).

Instead, introduce parentheses:

    git_log_cmd = (
        'git log -z ...'.format(...).split(' '))

If necessary, declare a constant for 'git log -z ...' and use that instead, in order to avoid the second backslash.